tests/unit/update-manager-test.js

```diff
@@ -198,4 +197,55 @@ describe('UpdateManager', () => {
 
   })
 
+  describe('editable', () => {
+    const self = {err: ''}
+    let updater, docuri, rterm, options, userCallback, loadStub
+    var loadStatus = 200
+
+    beforeEach(() => {
+      options = {}
+      userCallback = () => {}
+
+      updater = new UpdateManager()
+      updater.store.fetcher.webOperation = sinon.stub().resolves({ ok: true, status: 200, statusText: "Dummy stub 1"})
+      updater.store.fetcher.load = sinon.stub().resolves({ ok: true, status: 200, statusText: "Dummy stub 2"})
+
+    })
+
+    it('Should detect a document is editable from metadata', () => {
+      loadMeta(updater.store)
+      expect(updater.editable(doc1)).to.equal('SPARQL')
+    })
+
+    it('Should not detect a document is editable from metadata after flush', () => {
+      loadMeta(updater.store)
+      updater.flagAuthorizationMetadata()
+      expect(updater.editable(doc1)).to.equal(undefined)
+    })
+
+
+    it('Async version should detect a document is editable from metadata', async () => {
+      loadMeta(updater.store)
+      const result = await updater.checkEditable(doc1)
+      expect(result).to.equal('SPARQL')
+      expect(updater.editable(doc1)).to.equal('SPARQL')
+    })
+
+    it('Async version should not detect a document is editable from metadata after flush', async () => {
+      loadMeta(updater.store)
+
+      updater.flagAuthorizationMetadata()
+
+      expect(updater.editable(doc1)).to.equal(undefined)
+
+      const result = await updater.checkEditable(doc1)
+
+      expect(result).to.equal(undefined)
```

Comment:
What is the purpose of `outOfDate`. It appears to be either `undefined` or `true`

Reply:
The `outOfDate` is used to flag a response as outdated, so that the editable function does not return out of date information about whether a response is editable or not. E.g. if I fetched a resource before log in it was not editable. When I log in and fetch the resource again or call the async checkEditable() function it should respect the current information about editability.

Therefore I would like to see a proof (aka a test) that this is working, like that:

- given preloaded meta data says a resource is editable
- when I call editable() it says `SPARQL` (because of the metadata in the store)
- when I flagAuthorizationMetadata and call editable it says `undefined` (because of the `outOfDate` flag)
- when I call checkEditable it says (`SPARQL`) again (because current data is fetched and respected)

Reply:
> when I call checkEditable it says (SPARQL) again (because current data is fetched and respected)

This is possible by removing the flag in `checkEditable().`

```
   const meta = kb.fetcher.appNode
   const requests = kb.statementsMatching(undefined, this.ns.link('requestedURI'), kb.literal(uri), meta).map(st => st.subject)
   for (const request of requests) {
    // remove outOfDate metadata flag
    const response = kb.any(request, this.ns.link('response'), null, meta) as Quad_Subject
    if (response !== undefined) {
      const quad = kb.statementsMatching(response, this.ns.link('outOfDate'), true as any, meta)
      kb.remove(quad)
      
    }

```
Is it the expected behaviour without removing/reloading the document and the metadata ?
If `checkEditable(uri)` removes/reloads the document and the metadata => the test will need a fake `kb.fetcher.load(uri)`

I'm wondering what are the purpose of 
- `flagAuthorizationMetadata()`
- `checkEditable(uri)`

What is needed at login/logout ? is `flagAuthorizationMetaData()` enough ? shall a new store be reinitiated

Reply:
We could at least delete all the request/response meta data, not necessarily the whole store. I do not know why @timbl decided to go with a flag instead. We can discuss it in the call

Reply:
From the call: The out of date flag is used, to keep the data in case we need it (e.g. debugging)

Reply:
From the call: Since we mocked load, nobody adds the meta data, so the expecation is fine as it is